src/utils/uuidSubstring.ts

```diff
@@ -0,0 +1,17 @@
+import uuid from 'uuid/v4'
+
+/**
+ * Returns substring of UUID.
+ * Default is "xxxx-xxxx", where x is a hex char.
+ *
+ * @param {number} [indexStart=4] the start index
+ * @param {number} [indexEnd=9] the end index
+ *
+ * @return string
+ *
+ */
+
+const uuidSubstring = (indexStart: number = 4, indexEnd: number = 9) =>
```

Comment:
This will _probably_ be fine, but it is definitely going to increase the chance of collisions. According to [Wikipedia](https://en.wikipedia.org/wiki/Universally_unique_identifier), the substring of a UUID with range `4...9` is the second half of `time_low` and all of `time_mid`, omitting `time_hi_and_version`, `clock_seq_hi_and_res`, `clock_seq_low`, and `node`. I'll defer to @benadida on the cryptographic significance of this.

Reply:
thanks for the callout. Unfortunately, this won't work.

Chance of collision on a string is *much* higher than intuition. See the birthday paradox:
https://betterexplained.com/articles/understanding-the-birthday-paradox/

With 6 base-64 characters that's 36 bits, which means if those numbers were truly random, you'd get a 50% chance of a collision as soon as you have 2^18 ballots, aka 128,000 ballots. And with fewer ballots, still a healthy chance that it happens.

The accepted best practice to ensure no collisions ever is 128 bits, aka 16 bytes or 21 base-64 characters. I'd like to keep that for now.

Reply:
Thanks for the analysis gents. I'll break this into separate PRs for adding Ballot ID back and a second for a simpler, shorter, more human-friendly output.